### PR TITLE
Make Transformer 0.4.x compatible with 0.8.0 config (close #129)

### LINKED ISSUE #129: Make Transformer 0.4.x compatible with 0.8.0 config
Transformer 0.7.0+ has known performance issues and therefore when we upgrade pipelines to the new Snowflake Loader, the Transformer has to be downgraded to 0.4.2 using a hacky workaround (separate config files). 

Until we fix the performance issues with the Transformer we need a way to be sure that the Transformer without the performance issues can be deployed along with Loader latest version, without fudging the configs.  

For this we need the transformer 0.4.3 to be compatible with the new storage integration configs.

### PR DESCRIPTION
**Pull request by dilyand**
Not meant to be merged, just for review purposes.

### CODE REVIEW

**Comment by benjben** [comment]
Can we get some context @dilyand please ? https://github.com/snowplow-incubator/snowplow-snowflake-loader/issues/129 doesn't contain any

**Comment by dilyand** [comment]
> Can we get some context @dilyand please ? #129 doesn't contain any

I've added some details @benjben .

**Comment by dilyand** [comment]
Changed base branch to 0.4.2, because reviewing changes against `master` didn't make sense.

File: core/src/main/scala/com.snowplowanalytics.snowflake.core/Config.scala [hunk 1]
```diff
@@ -122,7 +123,11 @@ object Config {
         JInt(duration) <- fields.find(_._1 == "sessionDuration").map(_._2)
       } yield RoleAuth(roleArn, duration.toInt)
 
-      credentials.orElse(role).getOrElse(throw new MappingException("Cannot extract either RoleAuth or CredentialsAuth from auth"))
+      val storageIntegration = for {
+        JString(integrationName) <- fields.find(_._1 == "integrationName").map(_._2)
+      } yield StorageIntegration
+
+      credentials.orElse(role).orElse(storageIntegration).getOrElse(throw new MappingException("Cannot extract either RoleAuth or CredentialsAuth from auth"))
```
Comment (oguzhanunlu) [inline core/src/main/scala/com.snowplowanalytics.snowflake.core/Config.scala]: `or StorageIntegration` could be added to the message to cover all tried auth methods

File: .travis.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+dist: trusty
```
Comment (oguzhanunlu) [inline .travis.yml:1]: travis-ci.org will be shut down on Dec 31, in 10 days. I hope we can still publish this

File: core/src/test/scala/com/snowplowanalytics/snowflake/core/ConfigSpec.scala [hunk 3]
```diff
@@ -292,15 +292,13 @@ class ConfigSpec extends Specification {
...
       "--events-manifest", "eyJzY2hlbWEiOiJpZ2x1OmNvbS5zbm93cGxvd2FuYWx5dGljcy5zbm93cGxvdy5zdG9yYWdlL2FtYXpvbl9keW5hbW9kYl9jb25maWcvanNvbnNjaGVtYS8yLTAtMCIsImRhdGEiOnsibmFtZSI6ImxvY2FsIiwiYXV0aCI6eyJhY2Nlc3NLZXlJZCI6ImZha2VBY2Nlc3NLZXlJZCIsInNlY3JldEFjY2Vzc0tleSI6ImZha2VTZWNyZXRBY2Nlc3NLZXkifSwiYXdzUmVnaW9uIjoidXMtd2VzdC0xIiwiZHluYW1vZGJUYWJsZSI6InNub3dwbG93LWludGVncmF0aW9uLXRlc3QtY3Jvc3NiYXRjaC1kZWR1cGUiLCJpZCI6IjU2Nzk5YTI2LTk4MGMtNDE0OC04YmQ5LWMwMjFiOTg4YzY2OSIsInB1cnBvc2UiOiJFVkVOVFNfTUFOSUZFU1QifX0=").toArray
 
     val expected = CliTransformerConfiguration(
       Config(
-        auth = Config.CredentialsAuth(
```
Comment (chuwy) [inline core/src/test/scala/com/snowplowanalytics/snowflake/core/ConfigSpec.scala]: Do we still have a test with `CredentialsAuth`? Want to make sure it's still working.

**Review by chuwy — APPROVED** [review]

**Pull request closed without merging**